File: examquiz-backend/src/main/java/com/examquizai/backend/service/UserService.java
package com.examquizai.backend.service;

import com.examquizai.backend.dto.request.UpdateUserRequest;
import com.examquizai.backend.dto.response.UserResponse;

/**
 * User profile management operations.
 */
public interface UserService {

    UserResponse getCurrentUser(String email);

    UserResponse getUserById(String id);

    UserResponse updateCurrentUser(String email, UpdateUserRequest request);
}

File: examquiz-ai/frontend/src/services/userService.js
import { apiClient } from './apiClient';

export const userService = {
  async getCurrentUser() {
    const response = await apiClient.get('/api/v1/users/me');
    return response.data.data; // UserResponse
  },

  async updateCurrentUser({ fullName }) {
    const response = await apiClient.put('/api/v1/users/me', { fullName });
    return response.data.data; // UserResponse
  },
};
